relayer/chainproxy/tendermintRPC.go

```diff
@@ -428,30 +526,30 @@ func (nm *TendemintRpcMessage) Send(ctx context.Context, ch chan interface{}) (r
 		rpcMessage, err = rpc.CallContext(connectCtx, nm.msg.ID, nm.msg.Method, nm.msg.Params)
 	}
 
-	var replyMsg JsonrpcMessage
+	var replyMsg *RPCResponse
 	// the error check here would only wrap errors not from the rpc
 	if err != nil {
 		if strings.Contains(err.Error(), context.DeadlineExceeded.Error()) {
 			// Not an rpc error, return provider error without disclosing the endpoint address
 			return nil, "", nil, utils.LavaFormatError("Failed Sending Message", context.DeadlineExceeded, nil)
 		}
-
-		replyMsg = JsonrpcMessage{
-			Version: nm.msg.Version,
-			ID:      nm.msg.ID,
+		id, err := idFromRawMessage(nm.msg.ID)
+		if err != nil {
+			return nil, "", nil, utils.LavaFormatError("Failed parsing ID when getting rpc error", err, nil)
 		}
-		replyMsg.Error = &rpcclient.JsonError{
-			Code:    1,
-			Message: fmt.Sprintf("%s", err),
+		replyMsg = &RPCResponse{
+			JSONRPC: nm.msg.Version,
+			ID:      id,
+			Error:   convertErrorToRPCError(err),
```

Comment:
BUG: error here is always nil because you overwrite it with id, err := idFromRawMessage(nm.msg.ID) and bail out, did you mean to use the original error from the callContext?